ios/VisionCameraPluginBarcodeScanner.swift

```diff
@@ -47,16 +48,66 @@ class VisionCameraPluginBarcodeScanner: NSObject, FrameProcessorPluginBase {
     return barcodeData
   }
     
-  @objc
-  public static func callback(_ frame: Frame!, withArgs args: [Any]!) -> Any! {
-    let image = VisionImage(buffer: frame.buffer)
-    image.orientation = frame.orientation
-    let format = args[0] as? Int
-    let barcodeOptions = getScannerOptions(scannerFormat: format)
-    guard let barcodeResults = getScannerResults(image: image, barcodeScanner: BarcodeScanner.barcodeScanner(options: barcodeOptions)) else {
+    private static func getImageSize(sampleBuffer: CMSampleBuffer) -> CGSize {
+        let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+        let imageWidth  = CVPixelBufferGetWidth(imageBuffer!);
+        let imageHeight = CVPixelBufferGetHeight(imageBuffer!);
+        return CGSizeMake(CGFloat(imageWidth), CGFloat(imageHeight))
+    }
+
+    private static func scaleImage(sampleBuffer: CMSampleBuffer) -> CMSampleBuffer? {
+        let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer)!
+        let image = CIImage(cvPixelBuffer: imageBuffer)
+        
+        let scaleFilter = CIFilter(name: "CILanczosScaleTransform")
+        scaleFilter?.setValue(image, forKey: kCIInputImageKey)
+        scaleFilter?.setValue([1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7].randomElement(), forKey: kCIInputScaleKey)
+        scaleFilter?.setValue(1.0, forKey: kCIInputAspectRatioKey)
+        guard let scaledImage = scaleFilter?.outputImage else { return nil }
+
+        let context = CIContext()
+        context.render(scaledImage, to: imageBuffer)
+        let sampleBuffer: CMSampleBuffer? = createSampleBufferFrom(pixelBuffer: imageBuffer)
+        return sampleBuffer
+    }
+
+    private static func createSampleBufferFrom(pixelBuffer: CVPixelBuffer) -> CMSampleBuffer? {
+        var sampleBuffer: CMSampleBuffer?
+        var timimgInfo  = CMSampleTimingInfo()
+        var formatDescription: CMFormatDescription? = nil
+        
+        CMVideoFormatDescriptionCreateForImageBuffer(allocator: kCFAllocatorDefault, imageBuffer: pixelBuffer, formatDescriptionOut: &formatDescription)
+        
+        let osStatus = CMSampleBufferCreateReadyWithImageBuffer(
+          allocator: kCFAllocatorDefault,
+          imageBuffer: pixelBuffer,
+          formatDescription: formatDescription!,
+          sampleTiming: &timimgInfo,
+          sampleBufferOut: &sampleBuffer
+        )
+        
+        guard let buffer = sampleBuffer else {
+          print("Cannot create sample buffer")
           return nil
+        }
+        return buffer
+    }
+     
+    @objc
+    public static func callback(_ frame: Frame!, withArgs args: [Any]!) -> Any! {
```

Comment:
Is there any other type but `any` we can use? Wouldn't this just be a string or null? or I guess some object.

Reply:
> Is there any other type but `any` we can use? Wouldn't this just be a string or null? or I guess some object.

https://mrousavy.com/react-native-vision-camera/docs/guides/frame-processors-plugins-ios#manual-setup Just copied from the docs when creating it. but having it as 'any' isn't an issue in this case. as typescript validates the type of the arg as  'number' and gets  passed down to this native func and I convert it from 'any' to 'Int'